pyproject.toml

```diff
@@ -172,9 +179,32 @@ ignore = [
 ]
 
 [tool.pytest.ini_options]
+markers = [
+    "requires_jupytext",
+    "requires_black",
+    "requires_isort",
+    "requires_flake8",
+    "requires_autopep8",
+    "requires_nbconvert",
+    "requires_myst",
+    "requires_no_myst",
+    "requires_quarto",
+    "requires_pandoc",
+    "requires_no_pandoc",
+    "requires_sphinx_gallery",
+    "requires_user_kernel_python3",
+    "requires_ir_kernel",
```

Comment:
Instead of using these marks I would suggest using `pytest.importskip` and `try ... catch` with `import`/`pytest.skip` [^1]. `skip_on_windows` is also an easy one to absorb. `pre_commit` I'm 50/50 on it, for packaging I could definitely use it to skip, since those are never used outside of dev environments, but it might feel off when all of the other have automatic skip.

More useful marks would be if a test strictly requires internet connection, specific environments, etc. Was considering if the `unit`/`functional`/`integration` could require their own marks, but I think that would be overkill since they are already in separate folders.

[^1]: https://www.scivision.dev/pytest-skip-importerror/

Reply:
I see! I will come back to this later on this week.

Reply:
I have thought a bit more about these markers, and for now I would prefer to keep them. I think them explicit enough in their names (but agreed their implementation is not so easily accessible).